File: Week3/Wednesday/calculate-mode/python/calculate_mode.py
from typing import Counter


def calculate_mode(list_in):
    counted = {}
    for value in list_in:
        if value in counted:
            counted[value] += 1
        else:
            counted[value] = 1
    sorted_count = Counter(counted)
    results = []
    greatest = 0
    for m in sorted_count.most_common():
        if m[1] >= greatest:
            results.append(m[0])
            greatest = m[1]
        else:
            return results
        
    return results

File: Week3/Wednesday/calculate-mode/js/calculateMode.js
const calculate_mode = (arrayIn) => {
  let counted = {}
  for (value of arrayIn){
    if(value in counted){
      counted[value]++
    } else {
      counted[value] = 1
    }
  }
  let countedArray = Object.entries(counted)
  let sortedArray = countedArray.sort((a,b) => b[1] - a[1])
  let results = []
  let maxNum = 0
  for (count of sortedArray){
    if (count[1] >= maxNum){
      if ((/[0-9]/).test(count[0])) {
        results.push(Number(count[0]))
      } else {
        results.push(count[0])
      }
      maxNum = count[1]
    }
  }
  return results
}

module.exports = {
  calculate_mode
}